packages/rum-core/src/browser/cookieObservable.ts

```diff
@@ -0,0 +1,64 @@
+import type { Configuration, CookieStore } from '@datadog/browser-core'
+import {
+  setInterval,
+  clearInterval,
+  Observable,
+  addEventListener,
+  ONE_SECOND,
+  findCommaSeparatedValue,
+} from '@datadog/browser-core'
+
+export interface CookieStoreWindow extends Window {
+  cookieStore?: CookieStore
+}
+
+export type CookieObservable = ReturnType<typeof createCookieObservable>
+
+export type CookieChange = { name: string; value: string | undefined }
+
+export function createCookieObservable(configuration: Configuration, cookieName: string) {
+  return new Observable<CookieChange>(
+    (observable) =>
+      listenToCookieStoreChange(configuration, cookieName, (event) => observable.notify(event)) ??
+      watchCookieFallback(cookieName, (event) => observable.notify(event))
+  )
+}
+
+function listenToCookieStoreChange(
+  configuration: Configuration,
+  cookieName: string,
+  callback: (event: CookieChange) => void
+) {
+  if (!(window as CookieStoreWindow).cookieStore) {
+    return
+  }
+
+  const listener = addEventListener(configuration, (window as CookieStoreWindow).cookieStore!, 'change', (event) => {
+    event.changed
+      .concat(event.deleted)
+      .filter((change) => change.name === cookieName)
+      .forEach((change) => {
+        callback({
+          name: change.name,
+          value: change.value,
+        })
+      })
+  })
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏are we sure that we can't have ordering issues if a single event contains multiple changes of a single cookie?

Reply:
I tried with the both JS APIs but I never managed to get the same cookie twice in the same table. In fact, I got one change event for each API calls:
```js
cookieStore.addEventListener('change', (event) => {
  if ((event.changed || []).some((c) => c.name === 'datadog-ci-visibility-test-execution-id')) {
    console.log(event.changed)
   }
})

document.cookie = 'datadog-ci-visibility-test-execution-id=foo; SameSite=None; Secure'
document.cookie = 'datadog-ci-visibility-test-execution-id=bar; SameSite=None; Secure'
document.cookie = 'datadog-ci-visibility-test-execution-id=baz; SameSite=None; Secure'

cookieStore.set({
  name: 'datadog-ci-visibility-test-execution-id',
  value: 'foo'
})

cookieStore.set({
  name: 'datadog-ci-visibility-test-execution-id',
  value: 'bar'
})
```

Reply:
I could not find a clear answer to this question in the [spec](https://wicg.github.io/cookie-store/). So I asked for [clarification](https://github.com/WICG/cookie-store/issues/226). However I don't think it's a blocker for the current usage of the observable since the `datadog-ci-visibility-test-execution-id` will only be set once. Wdyt?

Reply:
Sure, it could be nice to add a comment in the implementation though.

Reply:
Done: https://github.com/DataDog/browser-sdk/blob/61410543b5e89c7e410bcd74dcff9c2d134f7309/packages/rum-core/src/browser/cookieObservable.ts#L37-L38